### PR TITLE
Add protobuff definition and an empty grpc service 

### LINKED ISSUE #2821: Helix Gateway Service
Support Helix Gateway Service for:

1. define protobuf for the service
2. implement liveness detection, session handle over.
3. implement failover mechanism

### PR DESCRIPTION
**Pull request by xyuanlu**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

resolves #2821 

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:
This PR adds a proto definition and an empty grpc service endpoint.

### Tests

- [X] The following tests are written for this issue:

(List the names of added unit/integration tests)
run mvn clean install

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-gateway/src/main/proto/HelixGatewayService.proto [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  }
+  string transitionID = 1;    // ID of transition message
+  TransitionType transitionType = 2;   // Transition type for shard operations
+  string resourceID = 3;              // Resource ID
+  string shardID = 4;                 // Shard to perform operation
+  string startState = 5;      // Shard start state
+  string targetState = 6;     // Shard target state.
+}
+
+message StateTransitionMessage{
```
Comment (junkaixue) [inline helix-gateway/src/main/proto/HelixGatewayService.proto:19]: We call single transition message previously, then following naming should be consistent. This should be called as TransitionMessages as well.

File: helix-gateway/src/main/proto/HelixGatewayService.proto [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  repeated SingleTransitionMessage request = 1;
+}
+
+message SingleReplicaStateTransitionStatus {
```
Comment (junkaixue) [inline helix-gateway/src/main/proto/HelixGatewayService.proto:23]: Shall we ignore replica in the naming? It is asymmetrical for naming in request/response. Also it is status of message also means for a replica already.

File: helix-gateway/src/main/java/org/apache/helix/gateway/grpcService/HelixGatewayServiceService.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package org.apache.helix.gateway.grpcService;
```
[the remark this answers is not in the document]
Comment (xyuanlu) [reply helix-gateway/src/main/java/org/apache/helix/gateway/grpcService/HelixGatewayServiceService.java:1]: TFTR will update

File: helix-gateway/src/main/proto/HelixGatewayService.proto (hunk 2 above)
Comment (xyuanlu) [reply helix-gateway/src/main/proto/HelixGatewayService.proto]: The outer request is names "replicaState", so I think probably we want to keep the replica in name. But we can simplify it as "SingleReplicaTransitionStatus"

**Review by junkaixue — APPROVED** [review]
Overall, lgtm. Just make sure the start state to be optional.

**Comment by xyuanlu** [comment]
The change is approved by @junkaixue // Thanks
Commit message 
Add protobuff definition and an empty grpc service

**Pull request merged**